src/parsers/manifest/dash/node_parsers/index.ts

```diff
@@ -215,6 +224,67 @@ const getLastLiveTimeReference = (
   return representationsMin;
 };
 
+/**
+ * From parsed media url, build a representation based media url.
+ * @param {Object} segmentIndex
+ * @param {string} repURL
+ * @param {string} repId
+ * @param {number} repBitrate
+ */
+function patchRepresentationIndex(
+  segmentIndex: IParsedSegmentList,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+): IListIndex;
+function patchRepresentationIndex(
+  segmentIndex: IParsedSegmentBase,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+): IBaseIndex;
+function patchRepresentationIndex(
+  segmentIndex: IParsedSegmentTemplate|IParsedSegmentTimeline,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+): ITemplateIndex|ITimelineIndex;
+function patchRepresentationIndex(
+  segmentIndex:
+    IParsedSegmentBase|IParsedSegmentList|IParsedSegmentTemplate|IParsedSegmentTimeline,
+  repURL: string,
+  repId: string,
+  repBitrate: number
+) {
```

Comment:
Isn't it a bit complicated?

If I read completely, for both initialization of regular segments' media, we only need:
 - some base URL
 - the URL in the MPD for the media
 - the ID of the representation
 - the bitrate of the representation

Doing a function which just replaces would be more lisible I think (and easier to define).